storage.go

```diff
@@ -366,49 +371,105 @@ func (f *file) Move(fromPath string, toPath string) FileResponse {
 	return response
 }
 
-// CreatSignedUrl create a signed url for a file object
-func (f *file) CreatSignedUrl(filePath string, expiresIn int) SignedUrlResponse {
-	_json, _ := json.Marshal(map[string]interface{}{
+func (f *file) CreateSignedURLForUpload(filePath string, expiresIn int) (*SignedURLForUploadResponse, error) {
+	reqBody, err := json.Marshal(map[string]interface{}{
 		"expiresIn": expiresIn,
 	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal request body: %w", err)
+	}
 
-	reqURL := fmt.Sprintf("%s/%s/object/sign/%s/%s", f.storage.client.BaseURL, StorageEndpoint, f.BucketId, filePath)
-	req, err := http.NewRequest(http.MethodPost, reqURL, bytes.NewBuffer(_json))
+	// Route for generating signed url for upload: /object/upload/sign/:bucketId/:objectKey
+	// See https://supabase.github.io/storage
+	reqURL := fmt.Sprintf("%s/%s/object/upload/sign/%s/%s", f.storage.client.BaseURL, StorageEndpoint, f.BucketId, filePath)
+	req, err := http.NewRequest(http.MethodPost, reqURL, bytes.NewBuffer(reqBody))
 	if err != nil {
-		panic(err)
+		return nil, fmt.Errorf("failed to create http request: %w", err)
```

Comment:
The correct error message would be "creating HTTP request...". But leave it as it is if it is already in multiple places.